### PR TITLE
test: add runMiniTest and some fixtures tests

### PR DESCRIPTION
**Pull request by JoelEinbinder**
`runMiniTest` defines a set of files inline and runs the test-runner on them.

### CODE REVIEW

**Review by mxschmitt — APPROVED** [review]
I would probably call it "runInlineTest" otherwise LGTM.

File: packages/test-runner/test/fixtures.spec.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      it('should use asdf', async ({asdf}) => {
+        expect(asdf).toBe(123);
+      });
+    `,
+  });
+  expect(results[0].status).toBe('passed');
+});
+
+
+it.skip('should work with a non-async function', async ({runInlineTest}) => {
```
Comment (pavelfeldman) [inline packages/test-runner/test/fixtures.spec.ts:33]: Let's never use skip, mark it as failing.

File: packages/test-runner/test/fixtures.spec.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      fixtures.defineTestFixture('asdf', async ({}, test) => await test(123));
+      it('should use asdf', ({asdf}) => {
+        expect(asdf).toBe(123);
+      });
+    `,
+  });
+  expect(results[0].status).toBe('passed');
+});
+
+it.skip('should fail with an unknown fixture', async ({runInlineTest}) => {
```
Comment (pavelfeldman) [inline packages/test-runner/test/fixtures.spec.ts:45]: ditto

**Review by pavelfeldman — APPROVED** [review]

File: packages/test-runner/test/fixtures.spec.ts (hunk 1 above)
Comment (JoelEinbinder) [reply packages/test-runner/test/fixtures.spec.ts:33]: What is the api to do that? I only had .only and .skip

Comment (pavelfeldman) [reply packages/test-runner/test/fixtures.spec.ts:33]: Check out how we do that in playwright.

**Pull request merged**